diesel/src/lib.rs

```diff
@@ -1,16 +1,95 @@
 //! # Diesel
 //!
 //! Diesel is an ORM and query builder designed to reduce the boilerplate for database interactions.
+//! If this is your first time reading this documentation,
+//! we recommend you start with the [getting started guide].
+//! We also have [many other long form guides].
 //!
-//! * [Homepage](https://diesel.rs/)
-//! * [Getting Started Guide](https://diesel.rs/guides/getting-started/)
-//! * [Source Code](https://github.com/diesel-rs/diesel/)
+//! [getting started guide]: https://diesel.rs/guides/getting-started/
+//! [many other long form guides]: https://diesel.rs/guides
 //!
-//! ## Crates
+//! # Where to find things
 //!
-//! * **Diesel**: core libary, documented here
-//! * [**Diesel CLI**](https://github.com/diesel-rs/diesel/tree/master/diesel_cli): command line tool that aids in managing your database schema
+//! ## Declaring your schema
 //!
+//! Diesel ensures that your queries are valid at compile time.
+//! This means it needs to know what your schema is.
+//! This is done using [the `table!` macro][`table!`].
+//! This macro is usually generated by `diesel print-schema` or [`infer_schema!`]
```

Comment:
Lots of "this". Maybe reword this block to something similar to...
> Diesel ensures that your queries are valid at compile time by knowing about your schema. This is done using  [the `table!` macro][`table!`], which is usually generated by `diesel print-schema` or [`infer_schema!`]

?

Reply:
Yeah. I want to keep the "it needs to know about" aspect, though.

> So Diesel is able to validate your queries at compile-time, it requires you to specify your database's schema in your code, which you can do with the `table!` macro. `diesel print-schema` or `infer_schema!` can be used to automatically generate these macro calls (by connecting to your database and parsing its schema).